src/v/cloud_storage_clients/client_pool.cc

```diff
@@ -459,10 +459,10 @@ bool client_pool::borrow_one(unsigned other) {
       other,
       _pool.size(),
       _capacity);
-    // TODO: do not use the topmost element. Find the one
+    // TODO: do not use the bottommost (oldest) element. Find the one
     // with expired connection.
-    auto c = _pool.back();
-    _pool.pop_back();
+    auto c = _pool.front();
```

Comment:
> cloud_storage_clients: destroy oldest client rather than newest on bo…
…rrow

please motivate the change

Reply:
I'm not sure this will help. It makes sense to cycle through all connections in the pool to avoid the expiration. If we will use something akin to LRU then we will increase the probability for the connection to get expired due to HTTP keep-alive. And the TLS handshake is a relatively heavy operation.